components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -878,32 +875,24 @@ private void setBaseUrlForRequestInformation(@Nonnull final RequestInformation r
             spanForAttributes.setAttribute(SemanticAttributes.SERVER_ADDRESS, requestURL.getHost());
             spanForAttributes.setAttribute(SemanticAttributes.URL_SCHEME, requestURL.getProtocol());
 
-            RequestBody body =
-                    requestInfo.content == null
-                            ? null
-                            : new RequestBody() {
-                                @Override
-                                public MediaType contentType() {
-                                    final Set<String> contentTypes =
-                                            requestInfo.headers.containsKey(contentTypeHeaderKey)
-                                                    ? requestInfo.headers.get(contentTypeHeaderKey)
-                                                    : new HashSet<>();
-                                    if (contentTypes.isEmpty()) {
-                                        return null;
-                                    } else {
-                                        final String contentType =
-                                                contentTypes.toArray(new String[] {})[0];
-                                        spanForAttributes.setAttribute(
-                                                "http.request_content_type", contentType);
-                                        return MediaType.parse(contentType);
-                                    }
-                                }
-
-                                @Override
-                                public void writeTo(@Nonnull BufferedSink sink) throws IOException {
-                                    sink.writeAll(Okio.source(requestInfo.content));
-                                }
-                            };
+            RequestBody body = null;
+            if (requestInfo.content != null) {
+                final Set<String> contentTypes =
+                        requestInfo.headers.containsKey(contentTypeHeaderKey)
+                                ? requestInfo.headers.get(contentTypeHeaderKey)
+                                : new HashSet<>();
+                final MediaType mediaType;
+                if (!contentTypes.isEmpty()) {
+                    String contentType = contentTypes.toArray(new String[] {})[0];
+                    spanForAttributes.setAttribute("http.request_content_type", contentType);
+                    mediaType = MediaType.parse(contentType);
+                } else {
+                    mediaType = null;
+                }
+                byte[] bytes = Compatibility.readAllBytes(requestInfo.content);
```

Comment:
Isn't this a potential problem for large files? Imagine a 30GB upload to a cloud drive. This would require a 30GB allocation, no?

Reply:
The reason I think could be a problem is because, for computers with lower RAM than the content size, the program will either crash (Allocation failure) or swap into disk if swapping is enabled (which will be a lot slower)

Reply:
yes this is going to cause memory pressure, which is going to cause the application to be at best slow at worst unstable. we should review our approach here.

Reply:
For the case of our SDK we have a limit of 5mb chunks, perhaps we can set this as the upper limit here as well?

Reply:
no we shouldn't, nothing prevents somebody from generating a client to an API that accepts a stream. And read that stream from the local file system (or from another network call). Limiting things here to 5MB would effectively block any scenario about 5MB.

For the SDK, and for others reading, we're referring to 5MB *slices* (chunk encoding is something different in HTTP) for the large file upload task, which is specific to Microsoft Graph.